llvm-general-pure/src/LLVM/General/AST/Operand.hs

```diff
@@ -16,18 +16,23 @@ newtype MetadataNodeID = MetadataNodeID Word
 
 -- | <http://llvm.org/docs/LangRef.html#metadata>
 data MetadataNode 
-  = MetadataNode [Maybe Operand]
+  = MetadataNode [Maybe Metadata]
   | MetadataNodeReference MetadataNodeID
   deriving (Eq, Ord, Read, Show, Typeable, Data)
 
+data Metadata
```

Comment:
I've tried to keep the haddock coverage of the exposed llvm-general API at 100%.

Please you give this type a haddock comment, ideally with linking into the llvm.org hosted language reference and/or source designed to clarify to the uninitiated user how the various llvm-general metadata types map to the non-haskell LLVM world. Often just a link is sufficient. For cases where the LLVM docs are themselves confusing, more is sometimes justified.